flatpak/org.nanuc.Axolotl.yml

```diff
@@ -58,20 +58,20 @@ modules:
         # tools.
         # (The directory format is: /run/build/MODULE_NAME/flatpak-node/cache)
         XDG_CACHE_HOME: /run/build/axolotl/flatpak-node/cache
-        # Set the npm cache directory, used by npm to find its package metadata.
-        npm_config_cache: /run/build/axolotl/flatpak-node/npm-cache
-        # Sets the directory where Node is located so that npm won't download the headers.
+        # Sets the directory where Node is located so that yarn won't download the headers.
         npm_config_nodedir: /usr/lib/sdk/node18
         # Cargo config
         CARGO_HOME: /run/build/axolotl/cargo
         # Cargo offline mode
         CARGO_NET_OFFLINE: "true"
     build-commands:
+      # Have Yarn use the offline mirror.
+      - HOME=$PWD yarn config --offline set yarn-offline-mirror $FLATPAK_BUILDER_BUILDDIR/flatpak-node/yarn-mirror
       # Install the packages from our offline cache.
-      - npm --prefix=axolotl-web --offline install
+      - yarn --cwd=axolotl-web --offline install
```

Comment:
It doesn't find the yarn config. Maybe try setting `HOME=$PWD` before every yarn command? Or is there something to explicitly specify the config file (like `--config` or similar)?

Reply:
I am a bit surprised too. Decided to try out a small demo in a separate project - https://gitlab.com/olof-nord/yarn-flatpak-demo/-/blob/main/com.demo.yarn-flatpak-demo.yml. Yarn should be on the path

```
➜  yarn-flatpak-demo git:(main) ✗ make debug-installed
/usr/bin/flatpak run --command=sh --devel com.demo.yarn-flatpak-demo
[📦 com.demo.yarn-flatpak-demo ~]$ ls -la /usr/lib/sdk/node18/bin
total 42472
drwxr-xr-x 2 nfsnobody nfsnobody     4096 Jan  1  1970 .
drwxr-xr-x 6 nfsnobody nfsnobody     4096 Jan  3 21:47 ..
lrwxrwxrwx 1 nfsnobody nfsnobody       45 Jan  3 21:47 corepack -> ../lib/node_modules/corepack/dist/corepack.js
-rwxr-xr-x 2 nfsnobody nfsnobody 43481624 Jan  1  1970 node
lrwxrwxrwx 1 nfsnobody nfsnobody       38 Jan  3 21:47 npm -> ../lib/node_modules/npm/bin/npm-cli.js
lrwxrwxrwx 1 nfsnobody nfsnobody       38 Jan  3 21:47 npx -> ../lib/node_modules/npm/bin/npx-cli.js
lrwxrwxrwx 1 nfsnobody nfsnobody       22 Jan  3 21:47 yarn -> ../share/yarn/bin/yarn
lrwxrwxrwx 1 nfsnobody nfsnobody       25 Jan  3 21:47 yarnpkg -> ../share/yarn/bin/yarnpkg
[📦 com.demo.yarn-flatpak-demo ~]$ 

```

Reply:
It's not about executable $PATH but about .yarnrc file which is relative to $HOME.

Reply:
I re-read your comment, I was too quick to reply indeed.

I have found some yarn Manifests which add HOME as an env variable, trying that out now.

https://github.com/flathub/com.logseq.Logseq/blob/master/com.logseq.Logseq.yml#L44